### PR TITLE
move labels to components which own the APIs

### PR DESCRIPTION
**Pull request by deads2k**
During the apimachinery split in 1.6, we accidentally moved several label APIs into apimachinery.  They don't belong there, since the individual APIs are not general machinery concerns, but instead are the concern of particular components: most commonly the kubelet.  This pull moves the labels into their owning components and out of API machinery.

@kubernetes/sig-api-machinery-misc @kubernetes/api-reviewers @kubernetes/api-approvers 
@derekwaynecarr  since most of these are related to the kubelet

### CODE REVIEW

**Comment by smarterclayton** [comment]
/approve

Lgtm but want a second look from someone.  These do not belong in API machinery for sure.

My only concern might be that people don't treat these as apis, but the package names seem clear.

File: pkg/kubelet/apis/well_known_labels.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+	LabelInstanceType = "beta.kubernetes.io/instance-type"
+
+	LabelOS   = "beta.kubernetes.io/os"
+	LabelArch = "beta.kubernetes.io/arch"
+
+	// When feature-gate for TaintBasedEvictions=true flag is enabled,
+	// TaintNodeNotReady would be automatically added by node controller
+	// when node is not ready, and removed when node becomes ready.
+	TaintNodeNotReady = "node.alpha.kubernetes.io/notReady"
```
Comment (derekwaynecarr) [inline pkg/kubelet/apis/well_known_labels.go]: kubelet already imports plugin/pkg/scheduler/algorithm so somewhere in there for now would work for me for taint related fields?

**Comment by deads2k** [comment]
@derekwaynecarr moved.  ptal.

File: pkg/kubelet/apis/well_known_labels.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	// TaintNodeUnreachable would be automatically added by node controller
+	// when node becomes unreachable (corresponding to NodeReady status ConditionUnknown)
+	// and removed when node becomes reachable (NodeReady status ConditionTrue).
+	TaintNodeUnreachable = "node.alpha.kubernetes.io/unreachable"
+
+	// When kubelet is started with the "external" cloud provider, then
+	// it sets this taint on a node to mark it as unusable, until a controller
+	// from the cloud-controller-manager intitializes this node, and then removes
+	// the taint
+	TaintExternalCloudProvider = "node.cloudprovider.kubernetes.io/uninitialized"
```
Comment (derekwaynecarr) [inline pkg/kubelet/apis/well_known_labels.go]: this is defined twice.

File: pkg/kubelet/apis/well_known_labels.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	// When feature-gate for TaintBasedEvictions=true flag is enabled,
+	// TaintNodeNotReady would be automatically added by node controller
+	// when node is not ready, and removed when node becomes ready.
+	TaintNodeNotReady = "node.alpha.kubernetes.io/notReady"
+
+	// When feature-gate for TaintBasedEvictions=true flag is enabled,
+	// TaintNodeUnreachable would be automatically added by node controller
+	// when node becomes unreachable (corresponding to NodeReady status ConditionUnknown)
+	// and removed when node becomes reachable (NodeReady status ConditionTrue).
+	TaintNodeUnreachable = "node.alpha.kubernetes.io/unreachable"
```
Comment (derekwaynecarr) [inline pkg/kubelet/apis/well_known_labels.go]: same here

File: pkg/kubelet/apis/well_known_labels.go (hunk 2 above)
Comment (deads2k) [inline pkg/kubelet/apis/well_known_labels.go]: how embarrassing.

**Comment by deads2k** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by deads2k** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by deads2k** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Pull request merged**